app/desmos/cmd/init.go

```diff
@@ -0,0 +1,32 @@
+package cmd
+
+// DONTCOVER
+
+import (
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	srvconfig "github.com/cosmos/cosmos-sdk/server/config"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	cosmosgenutilcli "github.com/cosmos/cosmos-sdk/x/genutil/client/cli"
+)
+
+// initCmd returns a command that initializes all files needed for Tendermint
+// and the respective application.
+func initCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
+	cmd := cosmosgenutilcli.InitCmd(mbm, defaultNodeHome)
+	defaultRun := cmd.RunE
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		clientCtx := client.GetClientContextFromCmd(cmd)
+		err := defaultRun(cmd, args)
+
+		// Set app.toml file
+		appConfig := srvconfig.DefaultConfig()
+		appConfig.StateSync.SnapshotInterval = 500
```

Comment:
Probably we need to also edit the sync infos to make sure they are compatible with state sync. From the `app.toml` file we can read

```toml
# snapshot-interval specifies the block interval at which local state sync snapshots are
# taken (0 to disable). Must be a multiple of pruning-keep-every.
snapshot-interval = 0
```

So we should change these values as well: 

```toml
# default: the last 100 states are kept in addition to every 500th state; pruning at 10 block intervals
# nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
# everything: all saved states will be deleted, storing only the current state; pruning at 10 block intervals
# custom: allow pruning options to be manually specified through 'pruning-keep-recent', 'pruning-keep-every', and 'pruning-interval'
pruning = "default"

# These are applied if and only if the pruning strategy is custom.
pruning-keep-recent = "0"
pruning-keep-every = "0"
pruning-interval = "0"
```

Maybe something like this works: 

```toml
pruning = "custom"

# These are applied if and only if the pruning strategy is custom.
pruning-keep-recent = "100"
pruning-keep-every = "500"
pruning-interval = "10"
```

This way we can make sure that even if the `default` values change, state sync will work properly